Source/DafnyCore/Resolver/PreType/PreTypeConstraints.cs

```diff
@@ -638,6 +653,10 @@ public enum CommonConfirmationBag {
         case CommonConfirmationBag.IsNewtypeBaseTypeLegacy:
           return pt.Decl is NewtypeDecl || pt.Decl.Name == "int" || pt.Decl.Name == "real";
         case CommonConfirmationBag.IsNewtypeBaseTypeGeneral:
+          if (familyDeclName is "set" or "iset" or "seq" or "multiset" or "map" or "imap" || pt.Decl is DatatypeDecl) {
```

Comment:
List of magic constants, scary ! Can you register these somewhere and query the registry instead?